addons/mail/static/src/core/common/thread_model.js

```diff
@@ -548,7 +548,12 @@ export class Thread extends Record {
                     !this.store.Thread.onlineMemberStatuses.includes(member.persona?.im_status)
             );
         },
-        sort: (m1, m2) => (m1.persona?.name < m2.persona?.name ? -1 : 1),
+        sort: (m1, m2) => {
+            if (m1.persona?.name === m2.persona?.name) {
+                return m1.id - m2.id;
+            }
+            return m1.persona?.name < m2.persona?.name ? -1 : 1;
+        },
```

Comment:
In master, could reuse `store.sortOnlineMembers` (and rename the function so it's just a general-purpose sorting of members). That way the computation is exactly the same